nannou/src/wgpu/texture/mod.rs

```diff
@@ -751,23 +751,7 @@ pub fn data_size_bytes(desc: &wgpu::TextureDescriptor) -> usize {
 
 /// Return the size of the given texture format in bytes.
 pub fn format_size_bytes(format: wgpu::TextureFormat) -> u32 {
-    use crate::wgpu::TextureFormat::*;
-    match format {
-        R8Unorm | R8Snorm | R8Uint | R8Sint => 1,
-
-        R16Uint | R16Sint | R16Float | Rg8Unorm | Rg8Snorm | Rg8Uint | Rg8Sint => 2,
-
-        R32Uint | R32Sint | R32Float | Rg16Uint | Rg16Sint | Rg16Float | Rgba8Unorm
-        | Rgba8UnormSrgb | Rgba8Snorm | Rgba8Uint | Rgba8Sint | Bgra8Unorm | Bgra8UnormSrgb
-        | Rgb10a2Unorm | Rg11b10Float => 4,
-
-        Rg32Uint | Rg32Sint | Rg32Float | Rgba16Uint | Rgba16Sint | Rgba16Float | Rgba32Uint
-        | Rgba32Sint | Rgba32Float => 8,
-
-        Depth32Float | Depth24Plus | Depth24PlusStencil8 => 4,
-
-        _ => unimplemented!(),
-    }
+    format.describe().block_size as u32
```

Comment:
Nice find! The docs mention this might only return the same thing on uncompressed textures (not sure if this is relevant for `nannou`) -- https://docs.rs/wgpu-types/0.7.0/wgpu_types/struct.TextureFormatInfo.html

Reply:
Ahhh good point! Hmmm it might be worth opening a dedicated issue and doing a review of the use of `format_size_bytes` throughout nannou and how compressed textures might affect handling certain cases in nannou's "extended" texture API. I haven't personally used compressed textures yet, but have been curious about them for a while, especially for use in real-time video decoding formats that support random access like HAP.